pkg/build/strategies/onbuild/onbuild.go

```diff
@@ -131,7 +146,8 @@ func (builder *OnBuild) Build(config *api.Config) (*api.Result, error) {
 
 	if len(opts.Name) > 0 {
 		if imageID, err = builder.docker.GetImageID(opts.Name); err != nil {
-			return nil, err
+			buildResult.BuildInfo = updater.UpdateFailureReason(buildResult.BuildInfo, api.ReasonGenericS2IBuildFailed)
```

Comment:
this one is ok as is for me.

For the bulk of these errors we can't give them enough information anyway, we need to log the error and the user needs to look in the build logs to understand what went wrong.

the case we're really trying to solve are the "you fat fingered your git repo url" type cases where they shouldn't have to dig through a log to see the git clone operation failed.